core/src/main/java/org/kohsuke/stapler/Function.java

```diff
@@ -322,15 +329,24 @@ protected MethodHandle handle() {
             return handle;
         }
 
+        @Override
         public Object invoke(StaplerRequest req, StaplerResponse rsp, Object o, Object... args) throws IllegalAccessException, InvocationTargetException {
-            Object[] arguments = new Object[args.length + 1];
-            arguments[0] = o;
-            System.arraycopy(args, 0, arguments, 1, args.length);
+            Object[] arguments;
+            if (Modifier.isStatic(m.getModifiers())) {
+                arguments = args;
+            } else {
+                arguments = new Object[args.length + 1];
+                arguments[0] = o;
+                System.arraycopy(args, 0, arguments, 1, args.length);
+            }
             try {
                 return handle().invokeWithArguments(arguments);
```

Comment:
Not clear to me why `handle` is `protected` and not `private`. Unreviewed change by @kohsuke in aca8c023adbc051c80a5e1d785ca76411ff9ed32.